cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/Scheduler.java

```diff
@@ -161,21 +163,32 @@ void addSchedules(Iterable<? extends ProgramSchedule> schedules)
    */
   List<ProgramSchedule> listSchedules(ProgramId programId) throws NotFoundException;
 
+  /**
+   * Retrieve all schedule records for a given namespace with additional properties user and artifact ID added
+   * in schedule properties.
+   *
+   * @param namespaceId the namespace for which to list the schedule records
+   * @param filter the filter to be applied on the result schedule records
+   * @return a list of schedule records for the namespace; never null
+   */
+  List<ProgramSchedule> listSchedulesWithUserAndArtifactId(NamespaceId namespaceId,
```

Comment:
any suggestion for a better name? I think `UserAndArtifactId` are necessary since they are added in the returned results. want to indicate that the returned `ProgramSchedule` are not the original ones